### PR TITLE
feat(pipeline): enable explain batch predictions

### PR DESCRIPTION
**Pull request by frascuchon**
This PR includes a new method that will apply explain over a prediction batch

### CODE REVIEW

File: src/biome/text/_model.py [hunk 1]
```diff
@@ -307,12 +309,48 @@ def predict_batch(
...
+            predictions = self.forward_on_instances(instances)
+        except Exception as error:
+            raise WrongValueError(
+                f"Failed to make predictions for '{input_dicts}'"
+            ) from error
+
+        explained_predictions = [
+            self.__build_explained_prediction(prediction, instance, n_steps)
+            for instance, prediction in zip(instances, predictions)
+        ]
```
Comment (dcfidalgo) [inline src/biome/text/_model.py:350]: I think this loop is not necessary. In the explain methods of the heads, we always embed the [instance in a batch](https://github.com/recognai/biome-text/blob/master/src/biome/text/modules/heads/classification/text_classification.py#L84). So the input to the `IntegratedGradients.attribute` is already a batch.
But let's leave this possible optimization to another PR (Maybe add an TODO here).

Comment (frascuchon) [reply src/biome/text/_model.py:350]: We will tackle in another PR. We need provide a common mechanism for prediction --> explain conversion if we accept explain prediction batches.
